### DISCUSSION TITLE
Release 0.10.0.alpha.1

### DISCUSSION THREAD

**Discussion by coaxial** [benjaminoakes/maid#240] [General]
I can't figure out how to have `release-please` understand we're trying to do a prerelease. I think it doesn't parse the version properly and only considers `\d+\.\d+\.\d+`. The only thing I could think of is to manually edit the `version.rb` file in the release PR but I don't know if it will also work for the CHANGELOG.md and the github release/tags.

I'm also hesitant to do the suggested empty commit on `master` just to trigger a release with `git commit --allow-empty -m 'chore: release 0.10.0.pre.1' -m "Release-As: 0.10.0.pre.1"` because if that doesn't work either the only way to "fix" it would be to force-push to `master` with all the headaches it can cause.

**Comment by coaxial** [comment]
Yes, there is a bug in `release-please` :(

The regex to parse versions is `/(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)(-(?<preRelease>[^+]+))?(\+(?<build>.*))?/`, which will only match `0.10.0-pre1` with `pre1` as the `preRelease` match. `0.10.0.pre.1` won't match anything after the `patch` group.

I've made a fork to experiment and I could get it to pickup the prerelase with `-pre1`: https://github.com/coaxial/maid/pull/5

**Comment by benjaminoakes** [comment]
As long as it is a prerelease from RubyGems' perspective, the format is
isn't particularly important to me. Is there any overlap between what this
tool does and what Ruby gems needs?

**Comment by coaxial** [comment]
I'm not certain. In any case, I'd rather we use conventional version numbering and rubygems recommends `0.10.0.alpha.1` rather than `0.10.0-alpha.1` (as `release-please` would make it).

I have an open PR to fix this: https://github.com/googleapis/release-please/pull/1887

**Reply by benjaminoakes** [reply]
Nice!

**Comment by coaxial** [comment]
Alright, `v0.10.0.pre.alpha.1` of the gem has been released and published! Feel free to test it :)
